pkg/sql/schemachanger/scbuild/internal/scbuildstmt/create_index.go

```diff
@@ -103,20 +140,20 @@ func CreateIndex(b BuildCtx, n *tree.CreateIndex) {
 			// initial primary key and include the union of all of the added and
 			// dropped columns.
 			if target == scpb.ToPublic {
-				source = t
+				idxSpec.primary = t
```

Comment:
This is strange. In my view `idxSpec` only pertains to this new secondary index and `primary` should never be set. Besides, you set it to `nil` soon after.